src/hooks/useViolations.ts

```diff
@@ -79,8 +79,28 @@ function useViolations(violations: TransactionViolation[]) {
                     }));
             }
 
+            // missingTag has special logic for policies with dependent tags, because only one violation is returned for all tags
+            // when no tags are present, so the tag name isn't set in the violation data. That's why we add it here
+            if (policyHasDependentTags && currentViolations[0]?.name === CONST.VIOLATIONS.MISSING_TAG && data?.tagListName) {
```

Comment:
Just a suggestion: `currentViolations[0]` is used several times so we can create a variable for this like `firstViolation`.